xarray/core/alignment.py

```diff
@@ -576,7 +604,7 @@ def align(self) -> None:
 
         if self.join == "override":
             self.override_indexes()
-        elif self.join == "exact" and not self.copy:
+        elif not self.copy and self.join == "exact":
```

Comment:
Nothing has changed here? It's usually nicer to minimize the diff as much as possible, that makes reviewing easier.

Reply:
This is definitely a leftover for when I was tinkering with the logic!